src/test/java/nl/jqno/equalsverifier/integration/operational/MultipleTypeEqualsVerifierTest.java

```diff
@@ -102,6 +127,22 @@ public void fail_whenCallingForPackage_givenTwoClassesInPackageAreIncorrect() {
             );
     }
 
+    @Test
+    public void fail_whenCallingForPackageRecursively_givenFourClassesInPackageAreIncorrect() {
+        ExpectedException
+                .when(() -> EqualsVerifier.forPackage(INCORRECT_PACKAGE, true).verify())
+                .assertFailure()
+                .assertMessageContains(
+                        "EqualsVerifier found a problem in 4 classes.",
+                        "IncorrectM",
+                        "IncorrectN",
+                        "IncorrectO",
+                        "IncorrectP",
+                        "Subclass: equals is not final.",
+                        "Reflexivity: object does not equal itself:"
+                );
+    }
```

Comment:
What's the difference between this test and the previous one?

Reply:
I just searched all the tests for the forPackage method and duplicate with the search in the subpackages

There was already :
- fail_whenVerifyingAPackageWithTwoIncorrectClasses
- fail_whenCallingForPackage_givenTwoClassesInPackageAreIncorrect

And I added :
- fail_whenVerifyingAPackageRecursivelyWithFourIncorrectClasses
- fail_whenCallingForPackageRecursively_givenFourClassesInPackageAreIncorrect

My tests check that the classes present in the sub-packages are found. So there are 4 classes instead of 2 that have problems.

I wanted to stay consistent with the existing tests. But it's true that these tests seem redundant. The only difference between them would be that one just checks the name of the classes while the other checks the name of the classes with their package.

Reply:
You're right, upon closer look I see that the redundancy was already there, and that you only tried to be consistent with what was already there. That's obviously my bad. Thanks for the elaborate answer though!